Tests/SentryTests/Protocol/SentryEventTests.swift

```diff
@@ -109,7 +109,6 @@ class SentryEventTests: XCTestCase {
 
     func testDecode_WithAllProperties() throws {
         // Arrange
-        SentryDependencyContainer.sharedInstance().dateProvider = TestCurrentDateProvider()
```

Comment:
Why is the date provider not needed anymore?

Reply:
The event uses the current date per default. You need to set the TestCurrentDateProvider so it uses a deterministic date. Before this PR the tests sometimes failed because of the missing milliseconds precision. I added it back to surface such potential issues again. But the major downside is that it will lead to not deterministic tests. As this PR fixes the underlying issue, I think it's better to go back to deterministic tests. So I added it again.